### ISSUE TITLE
Linux and Windows support

### ISSUE THREAD

**Issue by phimath** [florent37/Flutter-AssetsAudioPlayer#386]
I see this supports Mac, is there any plan to support Linux and Windows as well?

**Comment by florent37** [comment]
I don't know how to play music in windows and linux, maybe using C++
I am not personally able to develop it but If someone send me a pull request with this, it would be awesome :)

**Comment by phimath** [comment]
Yeah this would be out of my ability as well I think.   For anyone who comes along though and wants to give it a go, [this](https://github.com/alexmercerind/flutter_audio_desktop) might be worth looking at:

> C++. It calls native methods using dart's MethodChannel. It uses [miniaudio](https://github.com/mackron/miniaudio) from [David Reid](https://github.com/mackron). 
> 
> I wrote [this](https://github.com/alexmercerind/flutter_audio_desktop/blob/master/audioplayer/audioplayer.hpp) simple wrapper in C++ around his library to get this working.

**Comment by DomingoMG** [comment]
I would like this to be a reality, this plugin called @phimath uses Windows, Linux and allows to change the sound card.
https://github.com/alexmercerind/flutter_audio_desktop/issues/3

I'm going to see if I can collaborate

**Comment by alexmercerind** [comment]
Hi there @florent37 !

Your library is the best playback library that I have ever seen.

Now, I have enough things ready to get Windows & Linux support on your this plugin.

You can [dart_vlc](https://github.com/alexmercerind/dart_vlc) plugin as a dependency, to bring Windows [Done] & Linux [WIP] to add Windows support to this project.

I've also talked to ryanheise (maintainer of just_audio) at [this issue thread](https://github.com/ryanheise/just_audio/issues/103#issuecomment-799690113) to get Windows support on his audio plugin as well.
I'll try to fix/add more features as possible.

Thing to notice here is that, you might need to change to federated plugin model, as ryanheise has done to his plugin, he has discussed the license things on that issue thread (I don't really know about licensing etc.). Since, libVLC & libVLC++ are under LGPL 2.1, thus my project [dart_vlc](https://github.com/alexmercerind/dart_vlc) is under LGPL 2.1.

Greetings.
Hope you like what I did.

**Comment by yhsj0919** [comment]
[lychee_player](https://github.com/boyan01/lychee_player)   also support Windows & linux

**Comment by sail1369** [comment]
when is windows support coming?

**Comment by alexmercerind** [comment]
@sail1369 the day you support/contribute.
